### PR TITLE
Prepare MSI's handling

### PR DESCRIPTION
**Pull request by damcou**
Add some changes to the extension to be more easily overriden by the Algoia MSI external module.

### CODE REVIEW

File: Helper/Entity/ProductHelper.php [hunk 1]
```diff
@@ -217,18 +217,11 @@ public function getProductCollectionQuery(
...
         }
 
         /* @noinspection PhpUnnecessaryFullyQualifiedNameInspection */
-        $products = $products->addFinalPrice()
-            ->addAttributeToSelect('special_price')
-            ->addAttributeToSelect('special_from_date')
-            ->addAttributeToSelect('special_to_date')
-            ->addAttributeToSelect('visibility')
-            ->addAttributeToSelect('status');
+        $products = $this->addMandatoryAttributes($products);
```
Comment (nunomaduro) [inline Helper/Entity/ProductHelper.php]:
```suggestion
        $this->addMandatoryAttributes($products);
```

File: Helper/Entity/ProductHelper.php [hunk 2]
```diff
@@ -263,6 +256,28 @@ public function getProductCollectionQuery(
...
+    {
+        /** @var \Magento\Catalog\Model\ResourceModel\Product\Collection $products */
+        $products->addFinalPrice()
+            ->addAttributeToSelect('special_price')
+            ->addAttributeToSelect('special_from_date')
+            ->addAttributeToSelect('special_to_date')
+            ->addAttributeToSelect('visibility')
+            ->addAttributeToSelect('status');
+
+        return $products;
```
Comment (nunomaduro) [inline Helper/Entity/ProductHelper.php]: I think you need to return nothing here. Seems that we are mutating `$products`.

File: Helper/Entity/ProductHelper.php [hunk 3]
```diff
@@ -728,7 +743,7 @@ private function addImageData(array $customData, Product $product, $additionalAt
         return $customData;
     }
 
-    private function addInStock($defaultData, $customData, Product $product)
+    protected function addInStock($defaultData, $customData, Product $product)
```
Comment (nunomaduro) [inline Helper/Entity/ProductHelper.php]: There is a reason why we are altering the visibility of this method?

File: Helper/Entity/ProductHelper.php [hunk 4]
```diff
@@ -41,7 +41,7 @@ class ProductHelper
     private $eventManager;
     private $visibility;
     private $stockHelper;
-    private $stockRegistry;
+    protected $stockRegistry;
```
Comment (nunomaduro) [inline Helper/Entity/ProductHelper.php]: There is a reason why we need to alter the visibility here?

Comment (damcou) [reply Helper/Entity/ProductHelper.php]: To be overriden by the inventory module, as discussed ;)

File: Helper/Entity/ProductHelper.php (hunk 3 above)
Comment (damcou) [reply Helper/Entity/ProductHelper.php]: To be overriden by the inventory module, as discussed ;)

File: Helper/Entity/ProductHelper.php (hunk 1 above)
Comment (damcou) [reply Helper/Entity/ProductHelper.php]: true ! :)

File: Helper/Entity/ProductHelper.php (hunk 2 above)
Comment (damcou) [reply Helper/Entity/ProductHelper.php]: done

**Pull request merged**